app/tasks.py

```diff
@@ -1,95 +1,56 @@
-import json
 import os
+import io
 from PIL import Image
-from app import celery, s3
-from app.base.utils import update_property_images
+from app import celery, s3, redis_client
 from config import S3_BUCKET_CONFIG
 
 bucket = S3_BUCKET_CONFIG["S3_BUCKET"]
-s3_prop_image_dir = S3_BUCKET_CONFIG["PROP_ASSETS"]
-s3_temp_dir = S3_BUCKET_CONFIG["TEMP_DIR"]
-s3_user_image_dir = S3_BUCKET_CONFIG["USER_ASSETS"]
 
 
 @celery.task()
-def process_images(filename, tmp_dir, dir_to_save):
+def image_process(folder_name, s3_dir):
     """
-    This function processes the image files, using the PIL image library. The filename argument is
-    used to create writable binary object, the object is overwritten by the data of the image downloaded
-    from s3 using boto3 client. Then the image processed using PIL is uploaded back to s3 using boto3 client.
-    The name of this object the filename argument. The image is deleted in the temp dir for cleanup.
+    Resize the image files, using the PIL image library
     """
-    try:
-        img_obj = open(filename, "wb")
-        s3.download_fileobj(bucket, f"{s3_temp_dir + tmp_dir}{filename}", img_obj)
-        pil_image_obj = Image.open(filename)
-        pil_image_obj.thumbnail((800, 800))
-        pil_image_obj.save(filename)
-
-        with open(filename, "rb") as data_obj:
-            image_uploader(data_obj, s3_prop_image_dir + dir_to_save + filename)
-        os.remove(filename)
-        delete_img_objs(bucket, s3_temp_dir + tmp_dir, file_obj=filename)
-    except Exception as e:
-        return e
+    img_data = redis_client.hgetall(folder_name)
+    for img_name in img_data.keys():
+        byte_key_to_str = img_name.decode("utf-8")
+        # Get image byte object from redis with r_key as object key
+        byte_img_obj = redis_client.hget(folder_name, byte_key_to_str)
+        decoded = Image.open(io.BytesIO(byte_img_obj))
+        decoded.thumbnail((800, 800))
+        decoded.save(byte_key_to_str, format="JPEG")
```

Comment:
are we transforming every image to `JEPG` here?
this will distort image quality and appearance

Reply:
I have removed it, it won't be necessary for now.